### DISCUSSION TITLE
hdf.f90 : No such file or directory

### DISCUSSION THREAD

**Discussion by dipeshpandit12** [NASA-LIS/LISF#1609] [Q&A]
When I previously ran ./compile without installing HDF4 and HDF-EOS2  the compilation worked successfully. However, when I tried  configuring and compiling it with hdf4, it showed this error:

![image](https://github.com/user-attachments/assets/2af4f2fe-ce39-4ea0-ba64-852341db5345)

----------------------------------------------------------------------
These are all the files I have installed:

![image](https://github.com/user-attachments/assets/857afe81-b3ae-4010-8784-f537fa36326f)

-----------------------------------------------------------------------
I was trying to run [test case](https://nasa-lis.github.io/LISF/public_testcase_walkthrough/public_testcase_walkthrough.html) for LDT. However, it showed the error "

Error termination. Backtrace:
#0  0x155550f19302 in ???
#1  0x155550f19dc9 in ???
#2  0x155550f1b252 in ???
#3  0x5a7000 in ???
#4  0x7e87b2 in ???
#5  0x542c54 in ???
#6  0x53ee22 in ???
#7  0x5e30e2 in ???
#8  0x59d73e in ???
#9  0x5f39cf in ???
#10  0x62a996 in ???
#11  0x406df6 in ???
#12  0x155550386d84 in ???
#13  0x406e2d in ???
#14  0xffffffffffffffff in ??? "

Then I ran tail ldtlog.0000 command to get the error and it showed this message "Reconfigure LDT with HDF4 support, recompile, and run again ".

**Comment by jvgeiger** [comment]
Hello.  What files are in your hdf-hdf4.3.0 directory?

**Reply by dipeshpandit12** [reply]
I have attached the image file:
![image](https://github.com/user-attachments/assets/61a6cfb0-675c-411a-a047-95cd5c213aa3)

**Comment by jvgeiger** [comment]
And what is in the hdf4-hdf4.3.0/include subdirectory?

**Reply by dipeshpandit12** [reply]
![image](https://github.com/user-attachments/assets/5fb631da-99f5-4894-8eaf-c4ef45e5a8d5)

**Comment by jvgeiger** [comment]
You did not enable Fortran support when you compiled HDF4.  Please reinstall HDF4, either adding `--enable-fortran --disable-netcdf` to you `configure` command, or adding `-DHDF4_BUILD_FORTRAN=ON -DHDF4_ENABLE_NETCDF=OFF` to your `cmake` command.

It also seems that you are not running `make install` at the end to cleanly install just the lib and include files into a separate location.

**Reply by dipeshpandit12** [reply]
When recompiling, the LDT I didn't get the error with hdf.f90 , however, I came across another problem, what steps do I need to carryout?

![image](https://github.com/user-attachments/assets/d4b18cf4-01f9-42b1-8220-69d81a69bcb9)

**Reply by dipeshpandit12** [reply]
I wanted to follow up on my question as I haven't received a response yet. Any insights or help would be greatly appreciated. Thanks in advance!

**Comment by jvgeiger** [comment]
Hello.  Sorry, we can get to these questions only as time permits.

In your ldt directory, you need to edit the make/configure.ldt file.  Add `-fallow-argument-mismatch` to the definition of `FFLAGS`.  Then run `make -C make realclean`.  Then run `./compile`.

**Comment by dipeshpandit12** [comment]
I compiled ldt using the command above, but encountered the following error (see attached image). Additionally, here is the list of paths in the /make/configure.ldt file.

1. 
![image](https://github.com/user-attachments/assets/9e1125c9-5ba1-4c26-b5db-2a66e46a5ae0)
2. 
![image](https://github.com/user-attachments/assets/46c89245-51f4-4b04-9d8b-4165ed3e1f86)

**Comment by jvgeiger** [comment]
Hello.  There is an error in your LDT environment variables.  LDT_JPEG is for specifying the location of the JPEG library.  Set this only if you cannot use the system provided version of libjpeg.  LDT_OPENJPEG is for specifying the location of the OpenJPEG library, libopenjp2.so.  The Build Instrustions section of the LDT Users Guide has more information (https://nasa-lis.github.io/LISF/LDT_users_guide/LDT_users_guide.html)

First, you must fix your envrionment variables:
1. do not set LDT_JPEG
1. set LDT_OPENJPEG=/home/rab347/LISF/openjpeg-2.5.2/build

Second, go into your LISF/ldt directory and:
1. save a copy of your current make/configure.ldt file
1. run `./configure`
1. compare the new make/configure.ldt file and your saved copy.  Copy over the other changes that I helped you make.
1. run `make -C make realclean`
1. run `./compile`

**Reply by dipeshpandit12** [reply]
I have actually set LDT_OPENJPEG=/home/rab347/LISF/openjpeg-2.5.2/build  in ~/.bashrc file:
![image](https://github.com/user-attachments/assets/56bc13a7-d255-48b2-ad2e-5d0290574174)

There is no difference in the file. And even after following the given steps, I got the same error.

**Comment by jvgeiger** [comment]
By the way, if you want to make changes to the LISF source code, then you must go through this compilation process.  But if you want to run LISF as is, then you can try our AppImage version of LISF.  Click on Releases and look either the "LISF master (pre-releaes) image" or the "LISF Public 7.5.1 Release".

**Comment by jvgeiger** [comment]
Hello.  Sorry for the delay.  Would you please determine where your JPEG library, libjpeg.so, is installed?  I would expect it do be in either /usr/lib, /usr/lib64, /lib, or /lib64.
